be/src/exec/hdfs_scanner_orc.cpp

```diff
@@ -306,30 +306,65 @@ bool OrcRowReaderFilter::filterOnPickStringDictionary(
     return false;
 }
 
-Status HdfsOrcScanner::do_open(RuntimeState* runtime_state) {
-    RETURN_IF_ERROR(open_random_access_file());
-    auto input_stream = std::make_unique<ORCHdfsFileStream>(_file.get(), _file->get_size().value(),
-                                                            _shared_buffered_input_stream.get());
-    input_stream->set_lazy_column_coalesce_counter(_scanner_ctx.lazy_column_coalesce_counter);
-    input_stream->set_app_stats(&_app_stats);
-    ORCHdfsFileStream* orc_hdfs_file_stream = input_stream.get();
+Status HdfsOrcScanner::build_iceberg_delete_builder() {
```

Comment:
nit: The new framework could consider multiple scan ranges of a data file to reuse its matching deleted files.

Reply:
I will do it in next pr.